### PR TITLE
[SYCL-MLIR][NFC] Refactor MLIR-SYCL dialect headers and source files

### PR DESCRIPTION
**Pull request by victor-eds**
Following other dialects structure, the following changes are introduced:

- Headers (split SYCLOpsDialect.h and SYCLOpInterfaces.h):
  - SYCLOps.h: Interfaces and operations declarations;
  - SYCLOpsDialect.h: Just the dialect declaration;
  - SYCLOpsTraits.h: Traits declarations.
- Source files (split SYCLOpsDialect.cpp):
  - SYCLOps.cpp: Interfaces and operations definitions;
  - SYCLOpsDialect.cpp: Just the dialect definition;
  - SYCLOpsTraits.cpp: Traits definitions.

cgeist now does not include `.h.inc` files from mlir-sycl, just `.h` files.

Signed-off-by: Victor Perez <victor.perez@codeplay.com>

### CODE REVIEW

File: mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Copyright (C) Codeplay Software Limited
```
[the remark this answers is not in the document]
Comment (victor-eds) [reply mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.h:1]: I've removed them from all header files.

File: mlir-sycl/lib/Dialect/IR/SYCLTraits.cpp [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+mlir::LogicalResult mlir::sycl::verifySYCLGetRangeTrait(Operation *Op) {
+  return verifyGetSYCLTyOperation(cast<mlir::sycl::SYCLMethodOpInterface>(Op),
+                                  "range");
+}
+
+mlir::LogicalResult mlir::sycl::verifySYCLGetGroupTrait(Operation *Op) {
+  return verifyGetSYCLTyOperation(cast<mlir::sycl::SYCLMethodOpInterface>(Op),
+                                  "range");
```
[the remark this answers is not in the document]
Comment (victor-eds) [reply mlir-sycl/lib/Dialect/IR/SYCLTraits.cpp:181]: As this is an NFC, just refactoring, I'd say this bug should be fixed in a different PR. I'll address it right now if you agree.

File: mlir-sycl/lib/Dialect/IR/SYCLOps.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/SYCL/IR/SYCLOps.h"
+
+#include "mlir/IR/OpImplementation.h"
+#include "llvm/ADT/TypeSwitch.h"
+
```
Comment (whitneywhtsang) [inline mlir-sycl/lib/Dialect/IR/SYCLOps.cpp:13]: Add `using namespace mlir::sycl;`?

**Review by whitneywhtsang — APPROVED** [review]

File: mlir-sycl/lib/Dialect/IR/SYCLOps.cpp (hunk 3 above)
Comment (victor-eds) [reply mlir-sycl/lib/Dialect/IR/SYCLOps.cpp:13]: Thanks. I just didn't want to introduce many changes in this PR, as they'd be difficult to visualize (as the code is being moved around). I'll do this the following days.

**Comment by whitneywhtsang** [comment]
Please ignore the test suite failures for now, they are unrelated to your change. I am investigating them.

File: mlir-sycl/lib/Dialect/IR/SYCLTraits.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (etiotto) [reply mlir-sycl/lib/Dialect/IR/SYCLTraits.cpp:182]: Yes, is important to fix the problem. If you prefer to do it in a different PR that is fine or make this PR not "NFC".

**Pull request merged**